### DISCUSSION TITLE
How to use in code

### DISCUSSION THREAD

**Discussion by Krest01** [deepjyoti30/ytmdl#258] [Q&A]
I want to use ytmdl in project and dont know how an example should look. Couldn't find anything that isn't command line based.
Looking for something like this:
```
import youtube_dl
def run():
    video_url = input("please enter youtube video url:")
    video_info = youtube_dl.YoutubeDL().extract_info(
        url = video_url,download=False
    )
    filename = f"{video_info['title']}.mp3"
    options={
        'format':'bestaudio/best',
        'keepvideo':False,
        'outtmpl':filename,
    }

    with youtube_dl.YoutubeDL(options) as ydl:
        ydl.download([video_info['webpage_url']])

    print("Download complete... {}".format(filename))

if __name__=='__main__':
    run()

```

**Comment by deepjyoti30** [comment]
@Krest01 `ytmdl` is not aimed as a library (thus no docs regarding the API) though it can be used as one without any issue.

For starters, I would suggest you to go over the source code to find out the various functions that `ytmdl` provides. As a starting point, you can look at [`ytmdl/main.py`](https://github.com/deepjyoti30/ytmdl/blob/master/ytmdl/main.py). This is where the entry for the app is and from here you will understand the flow of `ytmdl`. Based on that you will also be able to find the functions that you want for your usecase.

There is a `core.py` file in the `ytmdl` directory. This file contains most of the top-level functions that `main.py` uses. This file can be thought of as the container for top-level pipeline stages where various stages are `searching`, `metadata searching`, `extracting`, `downloading` etc.

As I said, the best way to go about would be to look at the `main.py` and start from there.

Once you have gone through that, if you need any inputs on specific functions and how a particular function actually works, you can let me know, I will be happy to explain!
